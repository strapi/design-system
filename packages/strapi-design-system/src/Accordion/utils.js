export const getBackground = ({ expanded, disabled, variant }) => {
  let boxBackground;

  if (expanded) {
    boxBackground = 'primary100';
  } else if (disabled) {
    boxBackground = 'neutral150';
  } else if (variant === 'primary') {
    boxBackground = 'neutral0';
  } else {
    boxBackground = 'neutral100';
  }

  return boxBackground;
};
