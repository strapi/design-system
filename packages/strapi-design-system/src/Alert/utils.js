export const handleBackgroundColor = ({ theme, variant }) => {
  if (variant === 'danger') {
    return theme.colors.danger100;
  }

  if (variant === 'success') {
    return theme.colors.success100;
  }

  return theme.colors.primary100;
};

export const handleBorderColor = ({ theme, variant }) => {
  if (variant === 'danger') {
    return theme.colors.danger200;
  }

  if (variant === 'success') {
    return theme.colors.success200;
  }

  return theme.colors.primary200;
};

export const handleIconColor = ({ theme, variant }) => {
  if (variant === 'danger') {
    return theme.colors.danger700;
  }

  if (variant === 'success') {
    return theme.colors.success700;
  }

  return theme.colors.primary700;
};
