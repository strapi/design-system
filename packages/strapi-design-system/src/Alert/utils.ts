import { DefaultTheme } from 'styled-components';

import type { AlertVariant } from './Alert';

interface StyledProps {
  theme: DefaultTheme;
  $variant: AlertVariant;
}

export const handleBackgroundColor = (variant: AlertVariant): keyof DefaultTheme['colors'] => {
  switch (variant) {
    case 'danger':
      return 'danger100';
    case 'success':
      return 'success100';
    case 'warning':
      return 'warning100';
    default:
      return 'primary100';
  }
};

// border-color is always 1 shade darker than background-color
export const handleBorderColor = (variant: AlertVariant): keyof DefaultTheme['colors'] => {
  return handleBackgroundColor(variant).replace('100', '200') as keyof DefaultTheme['colors'];
};

export const handleIconColor = ({ theme, $variant }: StyledProps) => {
  if ($variant === 'danger') {
    return theme.colors.danger700;
  }

  if ($variant === 'success') {
    return theme.colors.success700;
  }

  if ($variant === 'warning') {
    return theme.colors.warning700;
  }

  return theme.colors.primary700;
};
