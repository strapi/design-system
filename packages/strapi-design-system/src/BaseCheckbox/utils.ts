import { BaseCheckboxSize } from './BaseCheckbox';

export const getCheckboxSize = ({ $size }: { $size: BaseCheckboxSize }) => {
  if ($size === 'M') {
    return '18px';
  }

  return '20px';
};
