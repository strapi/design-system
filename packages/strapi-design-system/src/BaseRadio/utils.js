export const getRadioSize = ({ size }) => {
  if (size === 'M') {
    return '18px';
  }

  return '20px';
};

export const getSelectedRadioSize = ({ size }) => {
  if (size === 'M') {
    return '10px';
  }

  return '12px';
};

export const getSelectedRadioPosition = () => {
  /**
   * This is computed using:
   * size = (radio height - radio selected height) / 2 - borderSize
   * For S, it looks like
   * size = (18 - 10) / 2 - 1
   * size = 3
   *
   * For L, it looks like
   * size = (20 - 12) / 2 -1
   * size = 3
   */
  return '3px';
};
