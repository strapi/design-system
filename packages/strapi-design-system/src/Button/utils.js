import { Text } from '../Text';
import { LIGHT_VARIANTS, VARIANTS } from './constants';

export const getVariantColorName = (variant) => {
  if (LIGHT_VARIANTS.includes(variant)) {
    return variant.substring(0, variant.lastIndexOf('-'));
  }
  if (variant === 'tertiary') {
    return 'neutral';
  }
  if (['default', 'secondary'].includes(variant) || !VARIANTS.includes(variant)) {
    return 'primary';
  }

  return variant;
};

export const getDisabledStyle = ({ theme }) => {
  return `
    border: 1px solid ${theme.colors.neutral200};
    background: ${theme.colors.neutral150};
    ${Text} {
      color: ${theme.colors.neutral600};
    }
    svg {
      > g, path {
        fill: ${theme.colors.neutral600};
      }
    }
  `;
};

export const getHoverStyle = ({ theme, variant }) => {
  if ([...LIGHT_VARIANTS, 'secondary'].includes(variant)) {
    return `
      background-color: ${theme.colors.neutral0};
    `;
  }
  if (variant === 'tertiary') {
    return `
      background-color: ${theme.colors.neutral100};
    `;
  }

  return `
    background: ${theme.colors[`${getVariantColorName(variant)}500`]};
  `;
};

export const getActiveStyle = ({ theme, variant }) => {
  if ([...LIGHT_VARIANTS, 'secondary'].includes(variant)) {
    return `
      background-color: ${theme.colors.neutral0};
      border: 1px solid ${theme.colors[`${getVariantColorName(variant)}600`]};
      ${Text} {
        color: ${theme.colors[`${getVariantColorName(variant)}600`]};
      }
      svg {
        > g, path {
          fill: ${theme.colors[`${getVariantColorName(variant)}600`]};
        }
      }
    `;
  }
  if (variant === 'tertiary') {
    return `
      background-color: ${theme.colors.neutral150};
    `;
  }

  return `
    background: ${theme.colors[`${getVariantColorName(variant)}600`]};
  `;
};

export const getVariantStyle = ({ theme, variant }) => {
  switch (variant) {
    case 'danger-light':
    case 'success-light':
    case 'secondary': {
      return `
          border: 1px solid ${theme.colors[`${getVariantColorName(variant)}200`]};
          background: ${theme.colors[`${getVariantColorName(variant)}100`]};
          ${Text} {
            color: ${theme.colors[`${getVariantColorName(variant)}700`]};
          }
          svg {
            > g, path {
              fill: ${theme.colors[`${getVariantColorName(variant)}700`]};
            }
          }
        `;
    }
    case 'tertiary': {
      return `
          border: 1px solid ${theme.colors.neutral200};
          background: ${theme.colors.neutral0};
          ${Text} {
            color: ${theme.colors.neutral800};
          }
          svg {
            > g, path {
              fill: ${theme.colors.neutral800};
            }
          }
        `;
    }
    default: {
      return `
          border: 1px solid ${theme.colors[`${getVariantColorName(variant)}600`]};
          background: ${theme.colors[`${getVariantColorName(variant)}600`]};
        `;
    }
  }
};

export const getIconPosition = ({ size }) => {
  if (size === 'S') {
    return '1px';
  }

  return '3px';
};
