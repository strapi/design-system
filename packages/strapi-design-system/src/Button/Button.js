import React from 'react';

import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import Loader from '@strapi/icons/Loader';
import { Typography } from '../Typography';
import { Box } from '../Box';
import { getDisabledStyle, getHoverStyle, getActiveStyle, getVariantStyle } from './utils';
import { VARIANTS, BUTTON_SIZES } from './constants';
import { BaseButton } from '../BaseButton';

const rotation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`;

const LoadingWrapper = styled.div`
  animation: ${rotation} 2s infinite linear;
  will-change: transform;
`;

const BoxFullHeight = styled(Box)`
  height: 100%;
`;

export const ButtonWrapper = styled(BaseButton)`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.buttonPrimary600};
  border: 1px solid ${({ theme }) => theme.colors.buttonPrimary600};
  height: ${({ theme, size }) => theme.sizes.button[size]};
  padding-left: ${({ theme }) => theme.spaces[4]};
  padding-right: ${({ theme }) => theme.spaces[4]};

  ${Box} {
    display: flex;
    align-items: center;
  }

  ${Typography} {
    color: ${({ theme }) => theme.colors.buttonNeutral0};
  }

  &[aria-disabled='true'] {
    ${getDisabledStyle}
    &:active {
      ${getDisabledStyle}
    }
  }
  &:hover {
    ${getHoverStyle}
  }
  &:active {
    ${getActiveStyle}
  }
  ${getVariantStyle}
  ${({ fullWidth }) =>
    fullWidth &&
    `
    display: inline-flex;
    justify-content: center;
    width: 100%;
  `}
`;

export const Button = React.forwardRef(
  ({ variant, startIcon, endIcon, disabled, children, onClick, size, loading, fullWidth, ...props }, ref) => {
    const isDisabled = disabled || loading;

    const handleClick = (e) => {
      if (!isDisabled && onClick) {
        onClick(e);
      }
    };

    return (
      <ButtonWrapper
        ref={ref}
        aria-disabled={isDisabled}
        disabled={isDisabled}
        size={size}
        variant={variant}
        onClick={handleClick}
        fullWidth={fullWidth}
        {...props}
      >
        {(startIcon || loading) && (
          <BoxFullHeight aria-hidden paddingRight={2}>
            {loading ? (
              <LoadingWrapper>
                <Loader />
              </LoadingWrapper>
            ) : (
              startIcon
            )}
          </BoxFullHeight>
        )}

        <Typography variant={size === 'S' ? 'pi' : undefined} fontWeight="bold" lineHeight={0}>
          {children}
        </Typography>

        {endIcon && (
          <Box aria-hidden paddingLeft={2}>
            {endIcon}
          </Box>
        )}
      </ButtonWrapper>
    );
  },
);

Button.displayName = 'Button';

Button.defaultProps = {
  disabled: false,
  endIcon: undefined,
  fullWidth: false,
  loading: false,
  onClick: undefined,
  size: 'S',
  startIcon: undefined,
  variant: 'default',
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  endIcon: PropTypes.element,
  fullWidth: PropTypes.bool,
  loading: PropTypes.bool,
  onClick: PropTypes.func,
  size: PropTypes.oneOf(BUTTON_SIZES),
  startIcon: PropTypes.element,
  variant: PropTypes.oneOf(VARIANTS),
};
