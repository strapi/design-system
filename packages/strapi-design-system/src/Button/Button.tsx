import React from 'react';

import { Loader } from '@strapi/icons';
import styled, { keyframes } from 'styled-components';

import { BUTTON_SIZES, Variant, ButtonSizes, DEFAULT } from './constants';
import { getDisabledStyle, getHoverStyle, getActiveStyle, getVariantStyle } from './utils';
import { BaseButton, BaseButtonProps } from '../BaseButton';
import { Box } from '../Box';
import { Flex } from '../Flex';
import { Typography } from '../Typography';

const rotation = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(359deg);
  }
`;

const LoaderAnimated = styled(Loader)`
  animation: ${rotation} 2s infinite linear;
  will-change: transform;
`;

export const ButtonWrapper = styled(BaseButton)<Required<Pick<ButtonProps, 'size' | 'variant'>>>`
  height: ${({ theme, size }) => theme.sizes.button[size]};

  svg {
    height: ${12 / 16}rem;
    width: auto;
  }

  &[aria-disabled='true'] {
    ${getDisabledStyle}

    &:active {
      ${getDisabledStyle}
    }
  }

  &:hover {
    ${getHoverStyle}
  }

  &:active {
    ${getActiveStyle}
  }

  ${getVariantStyle}
`;

export interface ButtonProps extends BaseButtonProps {
  endIcon?: React.ReactNode;
  fullWidth?: boolean;
  loading?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  size?: ButtonSizes;
  startIcon?: React.ReactNode;
  variant?: Variant;
}

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      variant = DEFAULT,
      startIcon,
      endIcon,
      disabled = false,
      children,
      onClick,
      size = BUTTON_SIZES[0],
      loading = false,
      fullWidth = false,
      ...props
    },
    ref,
  ) => {
    const isDisabled = disabled || loading;

    const handleClick: React.MouseEventHandler<HTMLButtonElement> = (e) => {
      if (!isDisabled && onClick) {
        onClick(e);
      }
    };

    return (
      <ButtonWrapper
        ref={ref}
        aria-disabled={isDisabled}
        disabled={isDisabled}
        size={size}
        variant={variant}
        onClick={handleClick}
        fullWidth={fullWidth}
        alignItems="center"
        background="buttonPrimary600"
        borderColor="buttonPrimary600"
        gap={2}
        inline={fullWidth}
        justifyContent={fullWidth ? 'center' : undefined}
        paddingLeft={4}
        paddingRight={4}
        width={fullWidth ? '100%' : undefined}
        {...props}
      >
        {(startIcon || loading) && <Box aria-hidden>{loading ? <LoaderAnimated /> : startIcon}</Box>}

        <Typography
          variant={size === 'S' ? 'pi' : undefined}
          fontWeight="bold"
          lineHeight={0}
          textColor="buttonNeutral0"
        >
          {children}
        </Typography>

        {endIcon && <Flex aria-hidden>{endIcon}</Flex>}
      </ButtonWrapper>
    );
  },
);

Button.displayName = 'Button';
