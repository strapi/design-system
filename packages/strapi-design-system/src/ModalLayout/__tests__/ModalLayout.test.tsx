import { render } from '@test/utils';

import { Button } from '../../Button';
import { ModalLayout, ModalHeader, ModalBody, ModalFooter } from '../index';

describe('ModalLayout', () => {
  it('should render component and match snapshot', () => {
    render(
      <ModalLayout onClose={() => jest.fn()} labelledBy="title">
        <ModalHeader>Modal Title</ModalHeader>
        <ModalBody>Hello World</ModalBody>
        <ModalFooter
          startActions={
            <Button onClick={() => jest.fn} variant="tertiary">
              Cancel
            </Button>
          }
          endActions={
            <>
              <Button variant="secondary">Add new stuff</Button>
              <Button onClick={() => jest.fn()}>Finish</Button>
            </>
          }
        />
      </ModalLayout>,
    );

    expect(document.body).toMatchSnapshot();
  });
});
