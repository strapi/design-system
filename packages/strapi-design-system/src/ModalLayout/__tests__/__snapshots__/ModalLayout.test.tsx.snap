// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalLayout should render component and match snapshot 1`] = `
.c0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c18 {
  font-size: 1.2rem;
  line-height: 1.33;
  font-weight: 600;
  color: #ffffff;
}

.c1 {
  padding-inline-end: 40px;
  padding-inline-start: 40px;
  position: fixed;
  z-index: 4;
}

.c4 {
  background: #ffffff;
  border-radius: 4px;
  box-shadow: 0px 2px 15px rgba(33,33,52,0.1);
  width: 83rem;
}

.c5 {
  background: #f6f6f9;
  padding-block-start: 16px;
  padding-inline-end: 20px;
  padding-block-end: 16px;
  padding-inline-start: 20px;
}

.c8 {
  background: #ffffff;
  padding-block: 8px;
  padding-inline: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
  cursor: pointer;
}

.c11 {
  padding-block: 32px;
  padding-inline: 32px;
}

.c15 {
  background: #4945ff;
  padding-block: 8px;
  padding-inline: 8px;
  padding-inline-end: 16px;
  padding-inline-start: 16px;
  border-radius: 4px;
  border-color: #4945ff;
  border: 1px solid #4945ff;
  cursor: pointer;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c9 {
  position: relative;
  outline: none;
}

.c9[aria-disabled='true'] {
  pointer-events: none;
}

.c9:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c9:focus-visible {
  outline: none;
}

.c9:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c16 {
  height: 3.2rem;
  border: 1px solid #dcdce4;
  background: #ffffff;
}

.c16 svg {
  height: 1.2rem;
}

.c16[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true'] .c17 {
  color: #666687;
}

.c16[aria-disabled='true'] svg > g,
.c16[aria-disabled='true'] svg path {
  fill: #666687;
}

.c16[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c16[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c16[aria-disabled='true']:active svg > g,
.c16[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c16:hover {
  background-color: #f6f6f9;
}

.c16:active {
  background-color: #eaeaef;
}

.c16 .c17 {
  color: #32324d;
}

.c16 svg > g,
.c16 svg path {
  fill: #32324d;
}

.c19 {
  height: 3.2rem;
  border: 1px solid #d9d8ff;
  background: #f0f0ff;
}

.c19 svg {
  height: 1.2rem;
}

.c19[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c19[aria-disabled='true'] .c17 {
  color: #666687;
}

.c19[aria-disabled='true'] svg > g,
.c19[aria-disabled='true'] svg path {
  fill: #666687;
}

.c19[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c19[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c19[aria-disabled='true']:active svg > g,
.c19[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c19:hover {
  background-color: #ffffff;
}

.c19:active {
  background-color: #ffffff;
  border: 1px solid #4945ff;
}

.c19:active .c17 {
  color: #4945ff;
}

.c19:active svg > g,
.c19:active svg path {
  fill: #4945ff;
}

.c19 .c17 {
  color: #271fe0;
}

.c19 svg > g,
.c19 svg path {
  fill: #271fe0;
}

.c20 {
  height: 3.2rem;
}

.c20 svg {
  height: 1.2rem;
}

.c20[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c20[aria-disabled='true'] .c17 {
  color: #666687;
}

.c20[aria-disabled='true'] svg > g,
.c20[aria-disabled='true'] svg path {
  fill: #666687;
}

.c20[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c20[aria-disabled='true']:active .c17 {
  color: #666687;
}

.c20[aria-disabled='true']:active svg > g,
.c20[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c20:hover {
  border: 1px solid #7b79ff;
  background: #7b79ff;
}

.c20:active {
  border: 1px solid #4945ff;
  background: #4945ff;
}

.c20 svg > g,
.c20 svg path {
  fill: #ffffff;
}

.c3 {
  background: #32324d1F;
  inset: 0;
}

.c10 {
  border-color: #dcdce4;
  height: 3.2rem;
  width: 3.2rem;
  color: #8e8ea9;
}

.c10:hover,
.c10:focus {
  color: #666687;
}

.c10[aria-disabled='true'] {
  color: #666687;
}

.c6 {
  border-radius: 4px 4px 0 0;
  border-bottom: 1px solid #eaeaef;
}

.c13 {
  border-radius: 0 0 4px 4px;
  border-top: 1px solid #eaeaef;
}

.c12 {
  overflow: auto;
  max-height: 60vh;
}

<body
  class="lock-body-scroll"
>
  <div>
    <div
      class="c0"
    >
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-log"
        role="log"
      />
      <p
        aria-live="polite"
        aria-relevant="all"
        id="live-region-status"
        role="status"
      />
      <p
        aria-live="assertive"
        aria-relevant="all"
        id="live-region-alert"
        role="alert"
      />
    </div>
  </div>
  <div
    class=""
  >
    <div
      class="c1 c2 c3"
    >
      <div>
        <div>
          <div
            aria-labelledby="title"
            aria-modal="true"
            class="c4"
            role="dialog"
          >
            <div
              class="c5 c6"
            >
              <div
                class="c7"
              >
                Modal Title
                <button
                  aria-disabled="false"
                  class="c8 c2 c9 c10"
                  type="button"
                >
                  <span
                    class="c0"
                  >
                    Close the modal
                  </span>
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="1.6rem"
                    viewBox="0 0 32 32"
                    width="1.6rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M26.061 23.939a1.503 1.503 0 0 1-2.125 2.125L16 18.125l-7.939 7.936a1.503 1.503 0 1 1-2.125-2.125L13.875 16 5.939 8.061a1.503 1.503 0 1 1 2.125-2.125L16 13.875l7.939-7.94a1.502 1.502 0 1 1 2.125 2.125L18.125 16z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="c11 c12"
            >
              Hello World
            </div>
            <div
              class="c5 c13"
            >
              <div
                class="c7"
              >
                <div
                  class="c14"
                >
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c16"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Cancel
                    </span>
                  </button>
                </div>
                <div
                  class="c14"
                >
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c19"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Add new stuff
                    </span>
                  </button>
                  <button
                    aria-disabled="false"
                    class="c15 c14 c9 c20"
                    type="button"
                  >
                    <span
                      class="c17 c18"
                    >
                      Finish
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
